### PR TITLE
Allow to subscribe on multiple routing keys on same queue

### PR DESCRIPTION
**Pull request by knapo**
Current format of `jackhammer.yaml` does not allow to subscribe on multiple routing keys on same queue, since the queue name is a hash key. E.g.

```
foo.bar:
  auto_delete: false
  durable: true
  exclusive: false
  handler: "BlueHandler"
  routing_key: "foo.bar.blue.#"
foo.bar:
  auto_delete: false
  durable: true
  exclusive: false
  handler: "OrangeHandler"
  routing_key: "foo.bar.orange.#" 
```

To achieve that queue config needs be an array e.g:

```
- queue_name: foo.bar
  auto_delete: false
  durable: true
  exclusive: false
  handler: "BlueHandler"
  routing_key: "foo.bar.blue.#"
- queue_name: foo.bar
  auto_delete: false
  durable: true
  exclusive: false
  handler: "OrangeHandler"
  routing_key: "foo.bar.orange.#"
```

### CODE REVIEW

File: spec/jackhammer/topic_manager_spec.rb [hunk 1]
```diff
@@ -35,7 +35,41 @@
...
+
+    context 'when Jackhammer.configuration.subscribe_yaml defines queues as array' do
+      let(:yaml) do
+        {
+          'my_topic' => {
+            'arguments' => true,
+            'auto_delete' => true,
+            'durable' => true,
+            'queues' => [
+              'queue_name' => 'first_queue',
```
Comment (xiazek) [inline spec/jackhammer/topic_manager_spec.rb]: 👍

**Comment by scottserok** [comment]
I might be misunderstanding the problem, but the current implementation allows for multiple queues under the same topic despite the hash configuration format.

The YAML file is loaded by `TopicManager`, and for each topic it loads the `queues` config (along with topic name and other options) into a new `Topic` object. These `Topic` objects are used to initiate the server subscriptions for each queue hash (handler, routing_key, and options) where the hash key is the name of the subscription. 

I'm not sure if the example provided is realistic, but if it is then there might be an issue with creating the subscriptions themselves. IIRC, it's important that subscription names (note: different than queue names) are unique because if there is more than one subscription with the same name on a queue, then only one of those subscriptions will receive a published event.

See `lib/jackhammer/topic.rb#24` where we iterate `@queue_config.map do |name, options|`. Perhaps the README needs to be updated, but I'm not convinced the current master can't handle this.  I opened https://github.com/RenoFi/jackhammer/pull/19 to double-check myself that we can have multiple subs configured.

**Comment by knapo** [comment]
I think I messed up `queue name` and `subscription name` things and the problem probably does not exist.

**Comment by knapo** [comment]
Closing and yanked 1.2.0.beta

**Pull request closed without merging**